src/components/ChannelPreview/hooks/useChannelPreviewDisplayAvatar.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect, useContext } from 'react';
+import { ChatContext } from '../../../context';
+
+/**
+ * Hook to set the display avatar for channel preview
+ * @param {*} channel
+ *
+ * @returns {object} e.g., { image: 'http://dummyurl.com/test.png', name: 'Uhtred Bebbanburg' }
+ */
+const useChannelPreviewDisplayAvatar = (channel) => {
+  const { client } = useContext(ChatContext);
+  const [displayAvatar, setDisplayAvatar] = useState({});
+
+  useEffect(() => {
+    if (channel.data.image) {
+      setDisplayAvatar({
+        image: channel.data.image,
+        name: channel.data.name,
+      });
+      return;
+    }
+
+    const members = channel.state ? Object.values(channel.state.members) : [];
```

Comment:
optional chaining. also just wondering but is state.members an object or an array? the name would suggest it's an array
```suggestion
    const members = Object.values(channel?.state?.members || {});
```